phoenix-core/src/it/java/org/apache/phoenix/end2end/InListIT.java

```diff
@@ -537,4 +539,267 @@ public void testInListExpressionWithDesc() throws Exception {
+        }
+
+
+        try (Connection viewConn = DriverManager.getConnection(TENANT_SPECIFIC_URL1)) {
+            viewConn.setAutoCommit(true);
+            Statement stmt = viewConn.createStatement();
+            stmt.execute("CREATE VIEW IF NOT EXISTS " + tenantView + " AS SELECT * FROM " + fullViewName);
+            viewConn.createStatement().execute("UPSERT INTO " + tenantView + "(ID1, ID2, EVENT_DATE) VALUES " +
+                    "('005xx000001Sv6o', '000000000000300', 1532458230000)");
+            viewConn.createStatement().execute("UPSERT INTO " + tenantView + "(ID1, ID2, EVENT_DATE) VALUES " +
+                    "('005xx000001Sv6o', '000000000000400', 1532458240000)");
+            viewConn.createStatement().execute("UPSERT INTO " + tenantView + "(ID1, ID2, EVENT_DATE) VALUES " +
+                    "('005xx000001Sv6o', '000000000000500', 1532458250000)");
+            viewConn.commit();
+
+            ResultSet rs = stmt.executeQuery("EXPLAIN DELETE FROM " + tenantView + " WHERE (ID1, ID2, EVENT_DATE) " +
+                    "IN (('005xx000001Sv6o', '000000000000400', 1532458240000)," +
+                    "('005xx000001Sv6o', '000000000000300', 1532458230000))");
+            assertTrue(getExplainPlanString(rs).contains("CLIENT PARALLEL 1-WAY RANGE SCAN"));
+
+            rs = stmt.executeQuery("EXPLAIN DELETE FROM " + tenantView + " WHERE (ID1, EVENT_DATE, ID2) IN " +
+                    "(('005xx000001Sv6o', 1532458240000, '000000000000400')," +
+                    "('005xx000001Sv6o', 1532458230000, '000000000000300'))");
+            assertTrue(getExplainPlanString(rs).contains("CLIENT PARALLEL 1-WAY RANGE SCAN"));
+
+            rs = stmt.executeQuery("EXPLAIN DELETE FROM " + tenantView + " WHERE (ID2, EVENT_DATE, ID1) IN " +
+                    "(('000000000000400', 1532458240000, '005xx000001Sv6o')," +
+                    "('000000000000300', 1532458230000, '005xx000001Sv6o'))");
+            assertTrue(getExplainPlanString(rs).contains("CLIENT PARALLEL 1-WAY RANGE SCAN"));
+
+            rs = stmt.executeQuery("EXPLAIN DELETE FROM " + tenantView + " WHERE (EVENT_DATE, ID2, ID1) IN " +
+                    "((1532458240000, '000000000000400', '005xx000001Sv6o')," +
+                    "(1532458230000, '000000000000300', '005xx000001Sv6o'))");
+            assertTrue(getExplainPlanString(rs).contains("CLIENT PARALLEL 1-WAY RANGE SCAN"));
+
+            stmt.execute("DELETE FROM " + tenantView + " WHERE (EVENT_DATE, ID2, ID1) IN " +
+                    "((1532458240000, '000000000000400', '005xx000001Sv6o')," +
+                    "(1532458230000, '000000000000300', '005xx000001Sv6o'))");
+
+            rs = stmt.executeQuery("SELECT ID2 FROM " + tenantView);
+            assertTrue(rs.next());
+            assertEquals("000000000000500", rs.getString(1));
+        }
+    }
+
+    @Test
+    public void testInListExpressionWithRightQueryPlan() throws Exception {
```

Comment:
Same comments as above, can also make these tests shared methods with a schema attribute to reuse code.

Reply:
the previous one tests DESC order by the PK, this one is ASC, which is not using the same schema.

Reply:
Right but what schema used can be an attribute as i mentioned.